sympy/combinatorics/fp_groups.py

```diff
@@ -527,38 +560,25 @@ def subgroup_quotient(G, H, fp_group=None):
     if isinstance(G, list) and not fp_group:
         raise ValueError("The fp_group must be"
                             "defined when the group is a list")
-    free_group = fp_group.G
+    free_group = fp_group.free_group
 
     if not isinstance(fp_group, FpGroup):
         raise ValueError("The parent group must be an instance"
                                     "of FpGroup")
 
-    if not (G.free_group == free_group
-                or H.free_group == free_group):
-        raise ValueError("The elements groups must belong to"
-                            "to the same free group as that of the parent group")
-
-    if isinstance(G, list):
-        # Todo - Set homomorphism to True after the merge
-        g_gens, g_rels, _gens = reidemeister_presentation(fp_group, G, rel=True)
-        _homomorphism = homomorphism(G, fp_group, g_gens, _gens, check=False)
-        g_gens = _gens
-        g_rels = _homomorphism(g_rels)
-    else:
-        g_gens = G.generators
-        g_rels = G.relators
-
-    if isinstance(H, list):
-        # Todo - Set homomorphism to True after the merge
-        h_gens, h_rels, _gens = reidemeister_presentation(fp_group, H, rel=True)
-        _homomorphism = homomorphism(H, fp_group, h_gens, _gens, check=False)
-        h_gens = _gens
-        h_rels = _homomorphism(h_rels)
-    else:
-        h_gens = H.generators
-        h_rels = H.relators
+    if ((isinstance(G, list) and not all(elem in free_group for elem in G))
+        or (isinstance(G, FpGroup) and not (G.free_group == free_group))):
+            raise ValueError("The elements groups must belong to"
+                                    "to the same free group as that of the parent group")
+    if ((isinstance(H, list) and not all(elem in free_group for elem in H))
+        or (isinstance(H, FpGroup) and not (H.free_group == free_group))):
+            raise ValueError("The elements groups must belong to"
+                                    "to the same free group as that of the parent group")
```

Comment:
Could be worth writing a little function `_check` that does this and then running
```
_check(G)
_check(H)
```
I think it would look a bit neater.

I would also suggest making the error message something like "The subgroup generators don't belong to the parent group". The fact that in SymPy, to belong to an `FpGroup` means the same as to belong to a `FreeGroup` from which it was constructed is a design choice, and might confuse a regular user who is not familiar with the implementation details